### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by TristanWright**
addresses #406 

- [ ] User documentation, how to use the application
  - [ ] Getting started
  - [x] Creating account
  - [x] Making trad clusters and ec2
  - [x] Making your first project and simulation
  - [x] Going through the simulation steps and running simulations.
- [ ] Developer documentation, setup, adding new workflows, important components
  - [x] Panels
  - [ ] Redux store, how each property is used and it's relation to others (if any)
  - [x] Making new workflows
  - [ ] Adding new tools

---
I wrote this up on a Chromebook while traveling and committed it using Github's file editor so there are certainly some typos and grammatical mistakes.

How does the structure look? A root README linking to folders and other documents?

### CODE REVIEW

**Comment by TristanWright** [comment]
@chetnieter I feel good about the documents in /usage if you would like to look at them.

Here's an easier way to browse them too: https://github.com/Kitware/HPCCloud/tree/docs/docs

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Table of Contents
+
+- [Getting Started](getting-started.md)
+- [Glossary](glossary.md)
+- Usage
+  - [Adding traditional clusters](/usage/trad-cluster.md)
+  - [Adding AWS profiles](/usage/aws-profiles.md)
+  - [Creating projects and simulations](/usage/creating.md)
+  - [Running jobs and simulations](/usage/running.md)
```
Comment (cjh1) [inline docs/README.md]: These links need updating

File: docs/dev-intro.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Delopment Introduction
+
```
Comment (cjh1) [inline docs/dev-intro.md]: Development

File: docs/dev-intro.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Delopment Introduction
```
Comment (chetnieter) [inline docs/dev-intro.md]: spelling: Development

File: docs/usage/trad-cluster.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Traditional Clusters
+
+Traditional clusters, or simply "clusters" in this document. Are machines that you can provision and run yourself.
```
Comment (chetnieter) [inline docs/usage/trad-cluster.md]: Should this be just one sentence?  "Traditional clusters, or simply "clusters" in this document, are machines that you can provision and run yourself."

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Table of Contents
+
+- [Getting Started](getting-started.md)
+- [Glossary](glossary.md)
+- Usage
```
Comment (chetnieter) [inline docs/README.md]: Do we want a section under usage about logging in or setting up an account?

File: docs/glossary.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+Workflows are collections of taskflows which define how you run simulations and jobs.
+
+### Projects
+
+Projects are collections of simulations.
+
+### Simulations
+
+Simulations are collections of workflows.
```
Comment (chetnieter) [inline docs/glossary.md]: Not sure I agree with this definition and seems somewhat circular with the definition of workflow.

File: docs/glossary.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+### Clusters
+
+Clusters are any machine that you can run tasks on.
+
+### Traditional Clusters
+
+Traditional clusters are machines that you have access and permissions for.
```
Comment (chetnieter) [inline docs/glossary.md]: Maybe a better definition would mention it corresponds to physical hardware and/or that fact it persists.

File: docs/usage/trad-cluster.md (hunk 4 above)
Comment (chetnieter) [inline docs/usage/trad-cluster.md]: Again not sure the definition of a traditional cluster is appropriate. Maybe something like: Traditional clusters typically involve physical hardware that dedicated to running simulations within the host operating system.

File: docs/usage/trad-cluster.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+### Testing
+
+Copy the shell command from the form field and paste it in a terminal. The command tries to connect to your cluster and if successful adds an ssh key to the cluster's key chain. This permits HPC-Cloud's backend (Cumulus) to connect to your cluster and run jobs and simulations. Click the "Test" button when the command is run and the key is saved on your cluster. If the test is successful cluster is ready to use in simulations.
+
+## Editing
+You can only edit the name of a cluster. If you need to change another detail delete and recreate the cluster.
+
+## Deleting
+
+You cannot delete clusters which are running simulations. With the cluster you want to delete selected, click "Delete cluster." You will be prompted before the cluster is deleted. You can always readd a deleted cluster, however the files generated from simulations will not be available through HPC-Cloud.
```
Comment (chetnieter) [inline docs/usage/trad-cluster.md]: Should 'readd' be 'read', 'ready', or something else?

File: docs/usage/aws-profiles.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+Click the "+" icon in the toolbar. You'll be presented a blank form in which you can fill out details of your AWS profile. Most important are AWS key and secret key (do not share these keys with anyone and especially don't commit them to a repository, much less this one). If necessary select a different machine type and region, the cheapest machine type per hour (Nano instance) and US-East(a) are selected by default. Click "Save Profile" and your AWS credentials will be validated, and if valid: saved. Your profile will be added to the list in the sidebar and you can now provision and launch EC2 instances from a Simulation Start view in a workflow.
+
+## Editing
+
+You cannot edit saved AWS profiles.
+
+## Deleting
+
+With the profile you want to delete selected, click "Delete profile." You will be prompted before the profile is deleted. This will not terminate still running instances associated with the profile. You will also not be able to access files generated from simulations which ran on instances using this profile, readding the profile will not allow you to do this either.
```
Comment (chetnieter) [inline docs/usage/aws-profiles.md]: readding -> reading ?

File: docs/usage/creating.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+### Editing 
+If you named or described your project incorrectly click on the pen icon on the homepage and modify these attributes. You cannot reupload files if the project required any.
+
+### Deleting
+ 
+You can only delete projects which have no simulations associated with them. Click on the pen icon for a project and you'll be taken to the edit page. Clicking the "Delete project" button will delete the project, you will be prompted beforehand. 
+
+## Simulations
+
+Simulations represent collections of taskflows depending on the type of parent project.
```
Comment (chetnieter) [inline docs/usage/creating.md]: In my mind a simulation is the process/act of running a simulation/modeling code.

Comment (TristanWright) [inline docs/usage/creating.md]: I think that is more like the Simulation _Step_, but it's also as you said. I'm thinking specifically when you "create simulation" you're making a set of steps/taskflows, one of which is the actual simulation which runs modeling code. Perhaps some term revision could be in order to distinguish Simulation, Simulation Step, and _Simulation_ (which runs modeling code)?

File: docs/README.md (hunk 5 above)
Comment (TristanWright) [inline docs/README.md]: Yes, I'll add it to a todo list here.

File: docs/usage/trad-cluster.md (hunk 4 above)
Comment (TristanWright) [inline docs/usage/trad-cluster.md]: I took your definition of Trad clusters verbatim, yes the opening should be one sentence. 👍

File: docs/glossary.md (hunk 6 above)
Comment (TristanWright) [inline docs/glossary.md]: By Gödel's ghost yes, it seems to be, or at least redundant. This could be mixed with my confusion on the definition of "Simulation" you brought up later.

Comment (TristanWright) [inline docs/glossary.md]: I think I've cleared up the workflow/taskflow confusion.

File: docs/usage/trad-cluster.md (hunk 8 above)
Comment (TristanWright) [inline docs/usage/trad-cluster.md]: re-add

File: docs/glossary.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
+# Terminology
+
+### Jobs 
+
+Jobs, or Primary Jobs, are special types of tasks run in simulations.
+
```
Comment (cjh1) [inline docs/glossary.md]: I think our definition of Jobs is really a Job running on an HPC resource?

File: docs/glossary.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+### Tasks
+
+Tasks are any thing run within simulations.
+
+### Taskflow
+
+Taskflows are a collection of tasks and jobs which are run for simulations.
+
```
Comment (cjh1) [inline docs/glossary.md]: Taskflows are collections of tasks run to orchestrate a give simulation workflow.

File: docs/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+## Development: 
+
+With the environment variable `DEVELOPMENT=1`, in your HPC-Cloud directory run:
+
+```
+$ npm start
+```
+
+This will run a webpack-dev server on `localhost:9999` which will reflect local changes to HPC-Cloud as you make them.
```
Comment (cjh1) [inline docs/getting-started.md]: This section is pretty much a duplicate of a part of the 'Development Introduction' so I think it should be move there.

File: docs/glossary.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+Projects are collections of simulations.
+
+### Simulations
+
+Simulations are collections of workflows.
+
+### Clusters
+
+Clusters are any machine that you can run tasks on.
+
```
Comment (cjh1) [inline docs/glossary.md]: s/task/jobs/

File: docs/glossary.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+### Traditional Clusters
+
+Traditional clusters are physical hardware dedicated to running simulations within the host operating system.
+
```
Comment (cjh1) [inline docs/glossary.md]: What about?

Traditional cluster are dedicated HPC resources, usually based on physical hardware.

File: docs/glossary.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+### AWS Profile
+
+AWS (Amazon Web Services) profiles consist of a Access key and a Secret key with which a user can provision EC2 instances.
+
+### EC2 Instances
+
+EC2 Instances are clusters, or groups of clusters, running in Amazon's EC2 Cloud.
+
```
Comment (cjh1) [inline docs/glossary.md]: EC2 instances usually refer to the individual virtual machines rather than a cluster.

File: docs/glossary.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+### ParaViewWeb
+
+ParaViewWeb is a tool in Paraview which allows a user to use ParaView through a web browser.
+
+### Simput
+
+A simulation deck input tool. Consult the Simput GitHub repository for more.
+
```
Comment (cjh1) [inline docs/glossary.md]: Link to SimPut repo?

File: docs/glossary.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+### Simulation step
+
+Steps of taskflows. They can have different states aswell. For example many steps have the substeps "Start" which starts a taskflow and "View" which views logs and statuses for the taskflow.
```
Comment (cjh1) [inline docs/glossary.md]: Simulation steps don't have a direct relationship to taskflows, so I would reword this.

File: docs/usage/aws-profiles.md [hunk 19]
```diff
@@ -0,0 +1,15 @@
+# AWS Profiles
+
+AWS profiles allow you to provision and run simulations on Amazon EC2 instances.
+
+## Creating 
+
+Click the "+" icon in the toolbar. You'll be presented a blank form in which you can fill out details of your AWS profile. Most important are AWS key and secret key (do not share these keys with anyone and especially don't commit them to a repository, much less this one). If necessary select a different machine type and region, the cheapest machine type per hour (Nano instance) and US-East(a) are selected by default. Click "Save Profile" and your AWS credentials will be validated, and if valid: saved. Your profile will be added to the list in the sidebar and you can now provision and launch EC2 instances from a Simulation Start view in a workflow.
```
Comment (cjh1) [inline docs/usage/aws-profiles.md]: Machine type is not a option when creating a profile. Its selected when a simulation is started on a new cluster.

File: docs/usage/aws-profiles.md [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+
+## Editing
+
+You cannot edit saved AWS profiles.
+
+## Deleting
+
+With the profile you want to delete selected, click "Delete profile." You will be prompted before the profile is deleted. This will not terminate still running instances associated with the profile. You will also not be able to access files generated from simulations which ran on instances using this profile, re-adding the profile will not allow you to do this either.
```
Comment (cjh1) [inline docs/usage/aws-profiles.md]: Its worth saying that you will be unable to delete a profile that is associated with an existing cluster. The cluster has to be deleted before the profile can be removed.

File: docs/usage/creating.md (hunk 10 above)
Comment (cjh1) [inline docs/usage/creating.md]: Taskflows are pretty incidental to this ( not sure we really want to expose the concept too much ), this important thing is they enable a set of tasks to be run that make up a particular simulation workflow.

File: docs/usage/mesh.msh [hunk 21]
```diff
@@ -0,0 +1,938 @@
...
+472 3 4 1 1 1 1 433 434 35 34
+473 3 4 1 1 1 1 434 435 36 35
+474 3 4 1 1 1 1 435 436 37 36
+475 3 4 1 1 1 1 436 437 38 37
+476 3 4 1 1 1 1 437 438 39 38
+477 3 4 1 1 1 1 438 439 40 39
+478 3 4 1 1 1 1 439 440 41 40
+479 3 4 1 1 1 1 440 441 42 41
+480 3 4 1 1 1 1 441 43 3 42
+$EndElements
```
Comment (cjh1) [inline docs/usage/mesh.msh]: Shouldn't we just link to the files in the PyFR repo? ( at least for the mesh )

File: docs/usage/running.md [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+1. Create a project and select the type 'pyfr-exec', name it whatever you wish. 
+2. Create a simulation, the input files mentioned above are required. Name the simulation whatever you wish.
+3. Go to your new simulation and click on the "Simulation" step in the left side bar.
+4. Here you can select your cluster or AWS Profile to use. If you've set the environment variable `DEMO=1` then the cluster you seek is under the "Traditional" cluster type option.
+5. Click "Run Simulation" and you'll be taken to the Simulation view sub-step. Here you can view logs and statuses of both tasks and jobs. When they all reach the "Complete" status your simulation has completed! A button "Visualize" will appear. Clicking on it will activate and take you to the disabled "Visualization" step in the sidebar.
+6. Here in the visualization step you will run Paraview web on the server of your choice to view the output of your simulation. If you'd like to run it on the same machine click "Start Visualization." You'll be taken to a job monitor page similar to the one for the Simulation step described in step 5.
```
Comment (cjh1) [inline docs/usage/running.md]: Rather than a server of your choice, it need to be the machine that simulation was run on ( currently )

File: docs/usage/running.md [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+7. Wait for the tasks to complete, and the primary job (ParaViewWeb) to be running. A button labeled "Visuzalize" will appear. Clicking this will take you to a ParaViewWeb tool in your browser where you can view and manipulate the output of your simulation. (wow!)
```
Comment (cjh1) [inline docs/usage/running.md]: Probably want to remove the (wow!) :-)

**Comment by cjh1** [comment]
@TristanWright I have added some comments, if you want I can push some changes directly on this branch.

File: docs/getting-started.md [hunk 24]
```diff
@@ -0,0 +1,18 @@
+# Getting Started
+
+Consult the full initial setup instructions on the [HPCCloud-deploy repository](https://github.com/Kitware/HPCCloud-deploy) and then come back here.
```
Comment (cjh1) [inline docs/getting-started.md]: I think we want to have step by step instructions here, even if it means some duplication:

1. Install prerequisites
2. git clone https://github.com/Kitware/HPCCloud-deploy
3. cd HPCCloud-deploy
4. DEMO=1 vagrant up

File: docs/usage/aws-profiles.md (hunk 19 above)
Comment (TristanWright) [inline docs/usage/aws-profiles.md]: Oops meant _availability zone_

File: docs/README.md [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+  - [Adding traditional clusters](usage/trad-cluster.md)
+  - [Adding AWS profiles](usage/aws-profiles.md)
+  - [Creating projects and simulations](usage/creating.md)
+  - [Running jobs and simulations](usage/running.md)
+- Development
+  - [Introduction](dev-intro.md)
+  - [Adding new workflows](#)
+    - [steps](#)
+    - [substeps](#)
```
Comment (cjh1) [inline docs/README.md]: @TristanWright Can we add some content for these sections.

Comment (TristanWright) [inline docs/README.md]: The development sections?

Comment (cjh1) [inline docs/README.md]: Yes, the ones that point to #

**Comment by cjh1** [comment]
@TristanWright  we add our logo, at least to the main README? Its all looking a little texty :sm

**Comment by TristanWright** [comment]
I seriously botched this rebase, @cjh1 can you push -f your local branch?

**Comment by cjh1** [comment]
I pushed what I have locally to docs-cjh1

**Comment by TristanWright** [comment]
wrong target branch
